src/web/app/components/question-types/question-edit-answer-form/mcq-question-edit-answer-form.component.ts

```diff
@@ -41,32 +41,32 @@ export class McqQuestionEditAnswerFormComponent
    * Updates the other option radio box when clicked.
    */
   updateIsOtherOption(): void {
-    this.responseDetails.isOther = !this.responseDetails.isOther;
+    this.triggerResponseDetailsChange('isOther', !this.responseDetails.isOther);
     this.isMcqOptionSelected[this.indexOfPreviousOptionSelected] = false;
     if (!this.responseDetails.isOther) {
-      this.responseDetails.otherFieldContent = '';
+      this.triggerResponseDetailsChange('otherFieldContent', '');
     } else {
-      this.responseDetails.answer = '';
+      this.triggerResponseDetailsChange('answer', '');
```

Comment:
It is a bit tricky here.

Because of the way `triggerResponseDetailsChange` is designed,
`this.triggerResponseDetailsChange('otherFieldContent', '');` won't see the change made in `this.triggerResponseDetailsChange('isOther', !this.responseDetails.isOther);`. That is to say; supposed `this.triggerResponseDetailsChange('isOther', !this.responseDetails.isOther);` change `isOther` from `true` to `false`. `this.triggerResponseDetailsChange('otherFieldContent', '');` will not observe the change of `isOther` and override it with stale value `true`.

There is a way to get rid of this. See `QuestionEditFormComponent#triggerModelChangeBatch`. Not sure whether it is a good way to do it. Any input @wkurniawan07 ?